app/Http/Controllers/TagsController.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace App\Http\Controllers;
+
+use App\Models\Tag;
+use App\Transformers\TagTransformer;
+
+class TagsController extends Controller
+{
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->middleware('require-scopes:public');
+    }
+
+    public function index()
+    {
+        $tags = \Cache::remember(
```

Comment:
Use `cache_remember_mutexed`.
Also it's probably fine to store in redis for now but depending on how large the list is and how frequently it's accessed, it might be better stored as a local cache singleton.